client/my-sites/pages/page/index.js

```diff
@@ -421,42 +430,154 @@ class Page extends Component {
 						siteUrl={ this.props.multisite && this.getSiteDomain() }
 					/>
 				</div>
-				{ ellipsisGridicon }
-				{ popoverMenu }
-				<ReactCSSTransitionGroup
-					transitionName="updated-trans"
-					transitionEnterTimeout={ 300 }
-					transitionLeaveTimeout={ 300 }
-				>
-					{ this.props.buildUpdateTemplate() }
-				</ReactCSSTransitionGroup>
+				{ ellipsisMenu }
+				{ this.renderShadowNotice() }
 			</CompactCard>
 		);
 	}
 
+	changeShadowStatus( shadowStatus ) {
+		return this.props.onShadowStatusChange( this.props.page.global_ID, shadowStatus );
+	}
+
+	async performUpdate( { action, progressNotice, successNotice, errorNotice, undo } ) {
+		await this.changeShadowStatus( progressNotice );
+		try {
+			await action();
+			if ( undo === 'undo' ) {
+				// This update was actually undo. Reset the shadow status immediately
+				await this.changeShadowStatus( false );
+				return;
+			}
+			await this.changeShadowStatus( { ...successNotice, undo } );
+			if ( undo ) {
+				// If undo is offered, keep the success notice displayed indefinitely
+				return;
+			}
+		} catch ( error ) {
+			await this.changeShadowStatus( errorNotice );
+		}
+		// remove the success/error notice after 5 seconds
+		await sleep( 5000 );
+		await this.changeShadowStatus( false );
+	}
```

Comment:
I think this async logic should live in an action creator instead of a component, but as I commented previously, that should be fine to happen in a separate PR that also implements this behavior for other post types.